### ISSUE TITLE
[iOS 12.4.6] Message from debugger: Terminated due to memory issue

### ISSUE THREAD

**Issue by calebmcelroy** [tanersener/react-native-ffmpeg#170]
Just wanna say thanks for the package it has been working great for other things I've been using it for. Hopefully I'm putting this issue in the right repository. I'm using react-native-ffmpeg but I know this is closely tied with mobile-ffmpeg.

**Description**
App crashes saying "Message from debugger: Terminated due to memory issue" no other indicators. Here the command I'm running:
```
FFmpeg.executeWithArguments([
  '-i', sourceUri,
  '-preset', 'faster',
  '-movflags', 'faststart',
  '-codec:a', 'aac',
  '-b:a', '128k',
  '-codec:v', 'libx264',
  '-pix_fmt', 'yuv420p',
  '-b:v', '3500k',
  '-minrate', '2000k',
  '-maxrate', '3500k',
  '-bufsize', '3500k',
  '-vf', 'scale=1080:-2',
  destUri,
])
```

_sourceUri_ is a .mov file and _destUri_ is a .mp4 file.

I tested on an android device that I have access to and it worked. It also works in the iPhone 8 Plus - 13.1 simulator running on OS X 10.14.6. But fails when running on iPhone 6 - iOS 12.4.6. I'm wondering if my device is running out of memory or if it's a bug?

**Expected behavior**
Encode the input video to a mp4 file with the desired bitrate, scale, & faststart for streaming.

**Current behavior**
It tries to start encoding but fails after a few frames according to logs

**Logs**
https://ghostbin.co/paste/y7yt5

**Environment**
_package.json:_
react-native: 0.61.3
react-native-ffmpeg: ^0.4.4

_Pod:_
mobile-ffmpeg-full-gpl (4.3.1.LTS)

_Device:_
iPhone 6 - iOS 12.4.6

**Comment by tanersener** [comment]
`react-native-ffmpeg` runs on top of `mobile-ffmpeg`. If you suspect a bug try testing different files on the same device, testing the same file on difference devices, using different `mobile-ffmpeg` versions/packages.

Besides, Xcode includes tools that you can use to monitor usage. You can use them to monitor memory usage and understand whether something is wrong about the library.

**Comment by calebmcelroy** [comment]
Thanks for your response. Upon further investigation I’ve determined that this error was due to my device running out of memory and not because of a bug. Testing on other devices with more memory, I determined encoding with my settings above required about ~300 MB of memory and my device only had ~200MB free so the memory pressure would cause the app to crash.

For anyone dealing with this same issue, I decreased the scale to 720p when the device’s total memory was 1GB or less, which reduced the memory usage enough to encode successfully.
